common/src/main/java/com/microsoft/identity/common/internal/cache/BrokerOAuth2TokenCache.java

```diff
@@ -402,6 +486,80 @@ public ICacheRecord load(@NonNull final String clientId,
         return resultRecord;
     }
 
+    /**
+     * The caller of this method should inspect the result carefully.
+     * <p>
+     * If the result contains >1 element: tokens were found for the provided filter criteria and
+     * additionally, tokens were found for this Account relative to a guest tenant.
+     * <p>
+     * If the result contains exactly 1 element, you may receive 1 of a few different
+     * response payloads, depending on cache state...
+     * <p>
+     * If the result contains an AccountRecord, IdTokenRecord, AccessTokenRecord, and
+     * RefreshTokenRecord then the result is OK to use. The caller should still check the expiry of
+     * the AccessTokenRecord before returning the result to the caller, refreshing as necessary...
+     * <p>
+     * If the result contains only an AccountRecord then we had no tokens in the cache and the
+     * library should do some equivalent of AUTH_REFRESH_FAILED_PROMPT_NOT_ALLOWED
+     * <p>
+     * If the result contains only an AccountRecord and RefreshTokenRecord then the caller should attempt to refresh
+     * the access token. If it works, call BrokerOAuth2TokenCache#save() with the result. If it
+     * fails, throw some equivalent of AUTH_REFRESH_FAILED_PROMPT_NOT_ALLOWED
+     *
+     * @param clientId The ClientId of the current app.
+     * @param target   The 'target' (scopes) the requested token should contain.
+     * @param account  The Account whose Credentials should be loaded.
+     * @return A List of ICacheRecords for the supplied filter criteria.
+     */
+    @SuppressWarnings(UNCHECKED)
+    @Override
+    public List<ICacheRecord> loadWithAggregatedAccountData(@NonNull final String clientId,
+                                                            @Nullable final String target,
+                                                            @NonNull final AccountRecord account) {
+        final String methodName = ":loadWithAggregatedAccountData";
+        final OAuth2TokenCache targetCache = getTokenCacheForClient(
+                clientId,
+                account.getEnvironment(),
+                mCallingProcessUid
+        );
+
+        final boolean shouldUseFociCache = null == targetCache;
+        final List<ICacheRecord> resultRecords;
+
+        if (shouldUseFociCache) {
+            // We do not have a cache for this app or it is not yet known to be a member of the family
+            // use the foci cache....
+
+            // Load a sparse-record (if available) containing only the desired account and a
+            // refresh token if available...
+            resultRecords = new ArrayList<>();
+            resultRecords.add(
+                    mFociCache.loadByFamilyId(
+                            clientId,
```

Comment:
Just qq. ClientId should not be necessary when fetching the FRT. Do you know why we need the clientId as a parameter here to load the foci cache?

Reply:
Is there any future extensibility you are considering here?

Reply:
This is only for [logging](https://github.com/AzureAD/microsoft-authentication-library-common-for-android/blob/dev/common/src/main/java/com/microsoft/identity/common/internal/cache/MicrosoftFamilyOAuth2TokenCache.java#L78-L81) (so we can see in logs which app is querying broker), the field is `@Nullable`; this is mentioned in the code comments [here](https://github.com/AzureAD/microsoft-authentication-library-common-for-android/blob/dev/common/src/main/java/com/microsoft/identity/common/internal/cache/MicrosoftFamilyOAuth2TokenCache.java#L88-L91)